src/utils.js

```diff
@@ -0,0 +1,123 @@
+const fs = require("fs");
+const read = require("fs-readdir-recursive");
+const { v4: uuidv4 } = require("uuid");
+
+function bundleForms() {
+  const formConfigPath = "./formSkeleton/";
+
+  let bundledFormOutput = fs.readFileSync(
+    formConfigPath + "inputFields/input-fields.ttl",
+    "utf8"
+  );
+
+  read(formConfigPath + "forms/").forEach((fileDir, i) => {
+    bundledFormOutput += fs.readFileSync(
+      formConfigPath + "forms/" + fileDir,
+      "utf8"
+    );
+  });
+
+  return bundledFormOutput;
+}
+
+function clearDistFolder() {
+  if (fs.existsSync("./dist")) {
+    fs.rmSync("./dist", { recursive: true });
+  }
+}
+
+function createFormTtlFile({ project, fileName, bundledFormOutput }) {
+  fs.mkdirSync(`./dist/${project.name}/config/semanticForms`, {
+    recursive: true,
+  });
+  fs.writeFileSync(
+    `./dist/${project.name}/config/semanticForms/${fileName}`,
+    bundledFormOutput
+  );
+}
+
+function createResourceMigrationFile({
+  project,
+  timestamp,
+  targetGraph,
+  fileName,
+  fileSize,
+  dateCreated,
+  dateModified,
+}) {
+  fs.mkdirSync(
+    `./dist/${project.name}/config/migrations/${timestamp}-add-new-forms`,
+    {
+      recursive: true,
+    }
+  );
+  fs.writeFileSync(
+    `./dist/${project.name}/config/migrations/${timestamp}-add-new-forms/${timestamp}-add-file-resource.sparql`,
+    generateMigrationContents({
+      targetGraph,
+      fileName,
+      fileSize,
+      dateCreated,
+      dateModified,
+    })
+  );
+}
+
+function generateMigrationContents({
+  targetGraph,
+  fileName,
+  fileSize,
+  dateCreated,
+  dateModified,
+}) {
+  let uuid = uuidv4();
+
+  // TODO: Should we use some sort of deindenting util instead of resetting the indentation here?
+  // prettier-ignore
+  return `\
+PREFIX nfo: <http://www.semanticdesktop.org/ontologies/2007/03/22/nfo#>
+PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+PREFIX dct: <http://purl.org/dc/terms/>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX dbpedia: <http://dbpedia.org/resource/>
+
+INSERT DATA{
+  GRAPH <${targetGraph}> {
+    <share://semantic-forms/${fileName}> a nfo:FileDataObject ;
+      dct:type <http://data.lblod.gift/concepts/form-file-type> ;
+      mu:uuid "${uuid}" ;
```

Comment:
Should the uuid be the same for all projects?

Reply:
I changed it to use the same uuid for all the projects since that probably makes the most sense.